twitter_api/main.py

```diff
@@ -16,6 +21,158 @@ def before_request():
 
 
 # implement your views here
+def query_db(query, args=(), one=False):
+    cur = g.db.execute(query, args)
+    rv = [dict((cur.description[idx][0], value)
+               for idx, value in enumerate(row)) for row in cur.fetchall()]
+    return (rv[0] if rv else None) if one else rv
+
+
+def get_user_id_from_token(token):
+    user = query_db('select user_id from auth where access_token=?', [token], one=True)
+    return user['user_id'] if user else None
+
+
+@app.route('/login', methods=['POST'])
+def login():
+    if request.data:
+        # retrieve post values
+        data = request.get_json()
+        username = data.get('username', None)
+        password = data.get('password', None)
+
+        # check if post info is complete
+        if username and password:
+            user = query_db('select * from user where username=?', [username], one=True)
+
+            # check if the user exists
+            if user:
+                if user['password'] != md5(password).hexdigest():
+                    # wrong password
+                    abort(401)
+
+                # check if they are logged in. don't generate duplicate token
+                token = query_db('select * from auth where user_id=?', [user['id']], one=True)
+
+                if not token:
+                    # generate the token
+                    query_db('insert into auth (user_id, access_token) values(?, ?)',
+                             [user['id'], create_token()])
+                    g.db.commit()
+                    # retrieve token
+                    token = query_db('select * from auth where user_id=?', [user['id']], one=True)
+
+                resp = {"access_token": token['access_token']}
+                resp = Response(response=json.dumps(resp), status=201, mimetype="application/json")
```

Comment:
can use `mimetype=JSON_MIME_TYPE` since it's a constant already defined in `utils.py`. also if you need to change the mimetype for everything, you only need to change it in one place